lib/wallet/account/hot-wallet-account.ts

```diff
@@ -34,7 +40,191 @@ export class HotWalletAccount implements IWallet {
+
+    const script = new ScriptBuilder()
+      .addData(publicKey.toXOnlyPublicKey().toString())
+      .addOp(Opcodes.OpCheckSig)
+      .addOp(Opcodes.OpFalse)
+      .addOp(Opcodes.OpIf)
+      .addData(Buffer.from("kasplex"))
+      .addI64(0n)
+      .addData(Buffer.from(JSON.stringify(payload, null, 0)))
+      .addOp(Opcodes.OpEndIf);
+
+    const P2SHAddress = addressFromScriptPublicKey(
+      script.createPayToScriptHashScript(),
+      this.networkId,
+    );
+
+    if (!P2SHAddress) {
+      throw new Error("Invalid P2SH address");
+    }
+
+    const { entries } = await this.rpcClient.getUtxosByAddresses({
+      addresses: [address.toString()],
+    });
+    const { transactions } = await createTransactions({
+      priorityEntries: [],
+      entries,
+      outputs: [
+        {
+          address: P2SHAddress.toString(),
+          amount: kaspaToSompi(Amount.ScriptUtxoAmount)!,
+        },
+      ],
+      priorityFee: kaspaToSompi(Amount.ScriptUtxoAmount),
+      changeAddress: address.toString(),
+      networkId: this.networkId,
+    });
+
+    for (const transaction of transactions) {
+      console.log(`Commit TX ID: ${transaction.id}`);
+      transaction.sign([privateKey]);
+      const hash = await transaction.submit(this.rpcClient);
+      submittedTxId = hash;
+    }
+
+    // Wait for the maturity event
+    const commitTimeout = setTimeout(() => {
+      if (!eventReceived) {
+        throw new Error("Commit transaction did not mature within 2 minutes");
+      }
+    }, 120000);
+
+    while (!eventReceived) {
+      await new Promise((resolve) => setTimeout(resolve, 500)); // wait and check every 500ms
+    }
+
+    clearTimeout(commitTimeout);
+
+    // Continue with reveal transaction after maturity event
+    eventReceived = false;
+    const { entries: newEntries } = await this.rpcClient.getUtxosByAddresses({
+      addresses: [address.toString()],
+    });
+
+    const revealUTXOs = await this.rpcClient.getUtxosByAddresses({
+      addresses: [P2SHAddress.toString()],
+    });
+
+    const { transactions: revealTransactions } = await createTransactions({
+      priorityEntries: [revealUTXOs.entries[0]],
+      entries: newEntries,
+      outputs: extraOutputs,
+      changeAddress: address.toString(),
+      priorityFee: revealPriorityFee,
+      networkId: this.networkId,
+    });
+    let revealHash: any;
+
+    for (const transaction of revealTransactions) {
+      console.log(`Reveal TX ID: ${transaction.id}`);
+      transaction.sign([privateKey], false);
+      const ourOutput = transaction.transaction.inputs.findIndex(
+        (input) => input.signatureScript === "",
+      );
+
+      if (ourOutput !== -1) {
+        const signature = transaction.createInputSignature(
+          ourOutput,
+          privateKey,
+        );
+        transaction.fillInput(
+          ourOutput,
+          script.encodePayToScriptHashSignatureScript(signature),
+        );
+      }
```

Comment:
This can be fully replaced by `signTx` method.

https://github.com/forbole/kastle/blob/a2516abac119500a0c22aa62fac0038b8290051f/lib/wallet/account/hot-wallet-account.ts#L341